File: scripts/Python/VowelsConsonant.py
def count_vowels_and_consonants(string):
    vowels = 0
    consonants = 0

    # Converting the string to lowercase for case-insensitive counting
    string = string.lower()

    for char in string:
        if char.isalpha():
            if char in "aeiou":
                vowels += 1
            else:
                consonants += 1

    return vowels, consonants

# Example usage:
input_string = input("Enter a string: ")
vowel_count, consonant_count = count_vowels_and_consonants(input_string)
print("Number of vowels:", vowel_count)
print("Number of consonants:", consonant_count)

File: scripts/C/VowelsConsonant.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

struct VowelConsonantCount {
    int vowels;
    int consonants;
};

struct VowelConsonantCount countVowelsAndConsonants(const char *str) {
    struct VowelConsonantCount counts;
    counts.vowels = 0;
    counts.consonants = 0;

    
    for (int i = 0; str[i] != '\0'; i++) {
        char c = tolower(str[i]); // Convert the character to lowercase

        // Check if the character is an alphabet letter
        if (isalpha(c)) {
            // Check if the character is a vowel
            if (strchr("aeiou", c) != NULL) {
                counts.vowels++;
            } else {
                counts.consonants++;
            }
        }
    }

    return counts;
}

int main() {
    
    const char *exampleString = "Hello World";
    struct VowelConsonantCount exampleResult = countVowelsAndConsonants(exampleString);

    printf("Example String - Vowels: %d\n", exampleResult.vowels);
    printf("Example String - Consonants: %d\n", exampleResult.consonants);

    return 0;
}
